db/models/models.js

```diff
@@ -7,11 +9,30 @@ exports.fetchAllTopics = () => {
   });
 };
 
-exports.fetchArticles = () => {
-  return db.query(`SELECT articles.* , COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments on comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;`).then((results) => {
-    return results.rows;
-  });
-};
+exports.fetchArticles = async (sortby = 'created_at', order = 'DESC', topics) => {
+  if (order !== 'DESC' && order !== 'ASC'){
+    return Promise.reject({ status: 404, msg: 'order not valid - must be ASC or DESC' });
+    }
+  if (sortby !== 'created_at' && sortby !== 'title' && sortby !== 'article_id' && sortby !== 'topic' && sortby !== 'author' && sortby !== 'body' && sortby !== 'votes'){
```

Comment:
could be nicer to make an 'accepted sorts' array of all the columns you'd accept and use an arr method to check if the given sortby is present in the arr - not a must-have but more something to consider to neaten the if statement up :D